src/pages/PagesRouter.tsx

```diff
@@ -44,7 +44,34 @@ export const PagesRouter: React.FC = () => {
         <HeaderWrapper />
         <Switch>
           <Route path='/swap' component={SwapPage} />
-          <Route path={'/newPosition'} component={NewPositionPage} />
+          <Route
+            path={'/newPosition/:item1?/:item2?/:item3?'}
+            render={({ match, history }) => {
+              let initialTokenFrom = ''
+              let initialTokenTo = ''
+              let initialFee = ''
+
+              if (match.params.item3) {
+                initialTokenFrom = match.params.item1 as string
+                initialTokenTo = match.params.item2 as string
+                initialFee = match.params.item3
+              } else if (match.params.item2) {
+                initialTokenFrom = match.params.item1 as string
+                initialFee = match.params.item2
+              } else if (match.params.item1) {
+                initialFee = match.params.item1
+              }
+
+              return (
+                <NewPositionPage
+                  initialTokenFrom={initialTokenFrom}
+                  initialTokenTo={initialTokenTo}
+                  initialFee={initialFee}
+                  history={history}
```

Comment:
When you navigate back in history, the path updates, but the UI remains unaffected. This results in an inconsistent state, especially when selecting different tokens, and the path indicates different ones. Is there a possibility that we can handle this?
![image](https://github.com/invariant-labs/webapp/assets/43304803/e0d413fa-68f3-4669-87ea-2d820a7d0b0f)